test/expr/print/protocol.swift

```diff
@@ -0,0 +1,24 @@
+// RUN: %target-swift-frontend -print-ast %s 2>&1 | %FileCheck %s
+
+protocol Archivable {
+  func archive(version: String)
+}
+// CHECK: internal protocol Archivable {
+// CHECK:   func archive(version: String)
+// CHECK: }
+
+func archive(_ a: Archivable) {
+  a.archive(version: "1")
+}
+// CHECK: internal func archive(_ a: Archivable) {
+// CHECK:   a.archive(version: "1")
+// CHECK: }
+
+func cast(_ a: Any) {
+  let conditional = a as? Archivable
+  let forced = a as! Archivable
+}
+// CHECK: internal func cast(_ a: Any) {
+// CHECK:   @_hasInitialValue private let conditional: Archivable? = a as? Archivable
+// CHECK:   @_hasInitialValue private let forced: Archivable = a as! Archivable
```

Comment:
I don't think we'll want to print access modifiers here

Reply:
It's really only used for debugging right now. If this was to be presented to a user, we'd want to filter this so it becomes:
```
let conditional = a as? Archivable
```
(removing `@_hasInitialValue private` and the explicit type annotation, if possible)

Reply:
Right yeah, my main concern was just that e.g:

```
func foo() {
  private let x = 0
}
```

isn't valid code